src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,104 @@
+package duke;
+
+/**
+ * Class that deals with making sense of the user command
+ */
+public class Parser {
+    /**
+     * Makes sense of the user command and calls corresponding methods
+     *
+     * @param userInput String of user input read from command line
+     * @param taskList TaskList object to which tasks are added to or deleted from
+     * @param fileManager Storage object that deals with loading tasks from file and saving tasks in file
+     */
+    public void parse(String userInput, TaskList taskList, Storage fileManager) {
```

Comment:
I like the way the parser is thought out here, but have perhaps using cases could make the code cleaner.